File: mcp/src/main/java/top/emilejones/hhu/mcp/entity/TextNode.java
package top.emilejones.hhu.mcp.entity;


import top.emilejones.hhu.mcp.enums.TextType;

public class TextNode {

    private String elementId;
    private Integer level;
    private Integer seq;
    private Integer name;
    private String text;
    private TextType type;

    // 构造函数
    public TextNode() {}

    public TextNode(String elementId, Integer level, Integer seq, Integer name, String text, TextType textType) {
        this.elementId = elementId;
        this.level = level;
        this.seq = seq;
        this.name = name;
        this.text = text;
        this.type = textType;
    }

    // getter & setter
    public String getElementId() {
        return elementId;
    }

    public void setElementId(String elementId) {
        this.elementId = elementId;
    }

    public Integer getLevel() {
        return level;
    }

    public void setLevel(Integer level) {
        this.level = level;
    }

    public Integer getSeq() {
        return seq;
    }

    public void setSeq(Integer seq) {
        this.seq = seq;
    }

    public Integer getName() {
        return name;
    }

    public void setName(Integer name) {
        this.name = name;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public TextType getType() {
        return type;
    }

    public void setType(TextType type) {
        this.type = type;
    }

    @Override
    public String toString() {
        return "TextNode{" +
                "elementId='" + elementId + '\'' +
                ", level=" + level +
                ", seq=" + seq +
                ", name=" + name +
                ", text='" + text + '\'' +
                ", type=" + type +
                '}';
    }
}

File: domain/src/main/kotlin/top/emilejones/hhu/domain/pipeline/TextNode.kt
package top.emilejones.hhu.domain.pipeline

import top.emilejones.hhu.domain.AggregateRoot

/**
 * 文本节点，承载切分后的文本内容和层级信息。
 * @author EmileJones
 */
class TextNode(
    override val id: String,
    val fileNodeId: String,
    val text: String,
    val seq: Int,
    val level: Int,
    val type: TextType,
    isEmbedded: Boolean,
    vector: List<Float>?
) : AggregateRoot<String>(id) {
    var vector: List<Float>? = vector
        private set
    var isEmbedded: Boolean = isEmbedded
        private set

    fun saveVector(vector: List<Float>) {
        require(!isEmbedded) { "TextNode[$id]节点向量已经被向量化" }
        require(this.vector == null) { "TextNode[$id]节点向量已经存在" }
        this.vector = vector
        this.isEmbedded = true
    }

}
